Obsidian/Net/Packets/Play/Clientbound/PlayerAbilitiesPacket.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Obsidian.Entities;
+
+namespace Obsidian.Net.Packets.Play.Clientbound;
+
+[Flags]
+public enum PlayerAbility
+{
+    None = 0x00,
+    Invulnerable = 0x01,
+    Flying = 0x02,
+    AllowFlying = 0x04,
+    CreativeMode = 0x08
+}
+
+public class PlayerAbilitiesPacket : IClientboundPacket, IServerboundPacket
+{
+    public PlayerAbility Abilities { get; set; } = PlayerAbility.None;
+
+    public float FlyingSpeed { get; set; } = 0.05F;
+
+    public float FieldOfViewModifier { get; set; } = 0.1F;
+
+    public int Id { get; }
+
+    public PlayerAbilitiesPacket(bool toClient)
+    {
+        Id = toClient ? 0x36 : 0x20;
+    }
+
+    public void Serialize(MinecraftStream stream)
+    {
+        using var packetStream = new MinecraftStream();
+        packetStream.WriteByte((byte)Abilities);
+        packetStream.WriteFloat(FlyingSpeed);
+        packetStream.WriteFloat(FieldOfViewModifier);
+
+        stream.Lock.Wait();
+        stream.WriteVarInt(Id.GetVarIntLength() + (int)packetStream.Length);
+        stream.WriteVarInt(Id);
+        packetStream.Position = 0;
+        packetStream.CopyTo(stream);
+        stream.Lock.Release();
+    }
+
+    public void Populate(MinecraftStream stream)
+    {
+        Abilities = (PlayerAbility) stream.ReadByte();
+    }
+
+    public void Populate(byte[] data)
+    {
+        using var stream = new MinecraftStream(data);
+        Populate(stream);
+    }
```

Comment:
Use the Field attribute on the packet properties so the source generators can write Serialize and Populate methods for you

Reply:
Apparently the SG does not like it because it does not have a parameter-less constructor.

Reply:
The Packet SG doesn't use constructors 🤨

Reply:
You have to make sure the class is set to partial ^ ^